src/login_nopam.c

```diff
@@ -263,18 +265,41 @@ static bool user_match (const char *tok, const char *string)
 
 static const char *resolve_hostname (const char *string)
 {
-	/*
-	 * Resolve hostname to numeric IP address, as suggested
-	 * by Dave Hagewood <admin@arrowweb.com>.  --marekm
-	 */
-	struct hostent *hp;
+	int                  gai;
+	char                 *p;
+	struct addrinfo      *addrs;
+	struct sockaddr      *sa;
+	struct sockaddr_in   *sin;
+	struct sockaddr_in6  *sin6;
 
-	hp = gethostbyname (string);
-	if (NULL != hp) {
-		return inet_ntoa (*((struct in_addr *) *(hp->h_addr_list)));
+	gai = getaddrinfo(string, NULL, NULL, &addrs);
+	if (gai != 0)
+		goto notfound;
+
+	sa = addrs[0].ai_addr;
+	switch (addrs[0].ai_addr->sa_family) {
```

Comment:
Just for readability, imo it would be nice if this used switch (sa->sa_family)

Reply:
Never mind - i see it doesn't matter :)

Reply:
Well, it can matter slightly for someone reading the diff, so I fixed it.  Anyway, it's a smaller diff now. :)